p2panda-rs/src/schema/schema_id.rs

```diff
@@ -10,25 +10,57 @@ use crate::hash::Hash;
 use crate::operation::Relation;
 use crate::schema::error::SchemaIdError;
 
+/// Struct representing a SchemaV1 id.
+#[derive(Clone, Debug, PartialEq)]
+pub struct SchemaV1(String);
+
+impl SchemaV1 {
+    fn as_str(&self) -> &str {
+        self.0.as_str()
+    }
+}
+
+impl Default for SchemaV1 {
+    fn default() -> Self {
+        Self("schema_v1".to_string())
+    }
+}
+
+/// Struct representing a SchemaFieldV1 id.
+#[derive(Clone, Debug, PartialEq)]
+pub struct SchemaFieldV1(String);
+
+impl SchemaFieldV1 {
+    fn as_str(&self) -> &str {
+        self.0.as_str()
+    }
+}
+
+impl Default for SchemaFieldV1 {
+    fn default() -> Self {
+        Self("schema_field_v1".to_string())
+    }
+}
+
 /// Identifies the schema of an [`crate::operation::Operation`].
 #[derive(Clone, Debug, PartialEq)]
 pub enum SchemaId {
     /// An application schema.
     Application(Relation),
 
     /// A schema definition.
-    Schema,
+    Schema(SchemaV1),
 
     /// A schema definition field.
-    SchemaField,
+    SchemaField(SchemaFieldV1),
 }
 
 impl SchemaId {
     /// Instantiate a new `SchemaId` from a hash string.
     pub fn new(hash: &str) -> Result<Self, SchemaIdError> {
         match hash {
-            "schema_v1" => Ok(SchemaId::Schema),
-            "schema_field_v1" => Ok(SchemaId::SchemaField),
+            "schema_v1" => Ok(SchemaId::Schema(SchemaV1::default())),
+            "schema_field_v1" => Ok(SchemaId::SchemaField(SchemaFieldV1::default())),
```

Comment:
Oh, this is interesting. What is the reason for using `Default` here?

Reply:
Well, not much reason really, except it felt nicer than just sticking a string in there. As the value will be the same for now (forever?)

Reply:
Ah!! Stupid me, I started off this PR with a different approach to the type coercion and this is leftover from that. I can remove all of the changes to `SchemaId` :roll_eyes:

Reply:
They're goooone

Reply:
Haha, alright! Yeah, I figured it would make sense if you would need the specific type of an schema enum variant :+1: